tests/syft/core/node/node_test.py

```diff
@@ -78,6 +85,30 @@ def test_register_vm_on_device_succeeds() -> None:
     assert bob_vm_client.device is not None
 
 
+def test_known_child_nodes() -> None:
+    bob_vm = sy.VirtualMachine(name="Bob VM")
+    bob_vm_client = bob_vm.get_client()
+    bob_vm.root_verify_key = bob_vm_client.verify_key  # inject 📡🔑 as 📍🗝
+
+    bob_vm_2 = sy.VirtualMachine(name="Bob VM 2")
+    bob_vm_client_2 = bob_vm_2.get_client()
+    bob_vm_2.root_verify_key = bob_vm_client_2.verify_key  # inject 📡🔑 as 📍🗝
+
+    bob_phone = sy.Device(name="Bob's iPhone")
+    bob_phone_client = bob_phone.get_client()
+    bob_phone.root_verify_key = bob_phone_client.verify_key  # inject 📡🔑 as 📍🗝
+
+    bob_phone_client.register(client=bob_vm_client)
+
+    assert len(bob_phone.known_child_nodes) == 1
+    assert bob_vm.address.vm in (obj.data.vm for obj in bob_phone.known_child_nodes)
```

Comment:
Q: Can't we simply check the address?
```bob_vm.address``` in bob_phone.known_child_nodes```?

Reply:
Good point!
I was having some problems because the adresses in known_child_nodes was composed of not only the vm but also the device... Not sure if it's a bug or what we want though.
What do you think about that?

Reply:
Yes sorry this is all a work around for the fact that we cant serialize clients so to ensure we can disptach in a node service message we keep a python object reference to the clients in a dictionary on the node and then look it up from its UID later. Like here:
```
# heritage_update_service.py
for node_client in node.known_child_nodes:
    try:
        # TODO: Client (and possibly Node) should subclass from StorableObject
        location_id = node_client.data.target_id.id
        msg.address = node_client.data
        try:
            in_memory_client = node.in_memory_client_registry[location_id]
```
 I am interested to know how we are supposed to do this properly so that we can make the changes now to avoid further confusion. One I idea I had was we could operate a a single localhost flask server (like duet) in solo mode and then post to localhost/uid for all the different "clients" so that way there are no references to clients hanging around only UID addresses and a loopback connector which they all listen to on their /UID url stub?

Reply:
I'm not 100% on all this part yet...
But why do we need to store the whole client? Couldn't we send a message only with the Address?